### PR TITLE
refactor(client): unify ws ping/pong API with the server

### PR DESCRIPTION
**Pull request by niklasad1**
This changes the clients to support a similar WebSocket ping/pong as the server which
was introduced in https://github.com/paritytech/jsonrpsee/pull/1248

The `PingConfig` type is duplicated in the server and client but I'm fine with that for now.

After introducing `Instant` in code I realized that the ping/pong interval would panic in wasm and I have removed ping/pongs completely because gloo-net doesn't support that anyway.

The other solution is to use some other crates where instants/interval stream are supported but it's not worth the effort IMHO and I think there was some reason in JS land where it's not possible to have client-side initiated pings.

### CODE REVIEW

File: core/src/client/async_client/mod.rs [hunk 1]
```diff
@@ -970,3 +1078,5 @@ impl<Fut: Future> Stream for MaybePendingFutures<Fut> {
 		self.futs.poll_next_unpin(cx)
 	}
 }
+
+
```
Comment (niklasad1) [inline core/src/client/async_client/mod.rs:1082]:
```suggestion
```

File: core/src/client/async_client/mod.rs [hunk 2]
```diff
@@ -831,53 +952,36 @@ where
...
 				}
-				_ = ping.tick() => {
+				_ = ping_interval.next() => {
 					if let Err(err) = sender.send_ping().await {
 						tracing::error!(target: LOG_TARGET, "Could not send ping frame: {err}");
 						break Err(Error::Custom("Could not send ping frame".into()));
 					}
 				}
 			}
-		}
```
Comment (jsdw) [inline core/src/client/async_client/mod.rs:841]: Nice to de-duplicate these bits :)

**Review by jsdw — APPROVED** [review]
Looks solid to me; good job!

File: core/src/client/async_client/mod.rs [hunk 3]
```diff
@@ -256,14 +370,21 @@ impl ClientBuilder {
 		S: TransportSenderT,
 		R: TransportReceiverT,
 	{
+		use futures_util::stream::Pending;
+
+		pub(crate) type PendingIntervalStream = IntervalStream<Pending<()>>;
```
Comment (niklasad1) [inline core/src/client/async_client/mod.rs:375]:
```suggestion
		type PendingIntervalStream = IntervalStream<Pending<()>>;		
```

File: core/src/client/async_client/mod.rs [hunk 4]
```diff
@@ -22,26 +50,92 @@ use helpers::{
...
+/// Configuration for WebSocket ping/pong mechanism and it may be used to disconnect
+/// an inactive connection.
+///
+/// jsonrpsee doesn't associate the ping/pong frames just that if
+/// pong frame isn't received within `inactive_limit` then it's regarded
+/// as missed.
+///
+/// Such that the `inactive_limit` should be configured to longer than a single
+/// WebSocket ping takes or it might be missed and may end up
+/// terminating the connection.
```
Comment (lexnv) [inline core/src/client/async_client/mod.rs:82]: nit: Might be worth stating the default values

**Review by lexnv — APPROVED** [review]
Nice one! 👍

**Pull request merged**